src/watcher/handler.py

```
@@ -192,6 +224,39 @@ def _handle_event(
+        ``OSError`` (e.g. the file was deleted before we could open it)
+        is treated as "allow" so transient races don't silence real events.
+        """
+        sd = self._safe_dir
+        if sd is None:
+            return True
+        try:
+            from utils.safedir import SymlinkRejected
+
+            fd = sd.open_child(path.name)
+            os.close(fd)
+            return True
```

Ruff (0.15.13)

[warning] Consider moving this statement to an `else` block

(TRY300)